src/core/immersed_boundary/ibm_tribend.cpp

```diff
@@ -40,45 +40,33 @@ void IBM_Tribend_CalcForce(Particle *p1, Particle *p2, Particle *p3,
   assert(p4);
 
   // Get vectors making up the two triangles
-  double dx1[3], dx2[3], dx3[3];
-  get_mi_vector(dx1, p1->r.p, p3->r.p);
-  get_mi_vector(dx2, p2->r.p, p3->r.p);
-  get_mi_vector(dx3, p4->r.p, p3->r.p);
+  auto const dx1 = get_mi_vector(p1->r.p, p3->r.p);
+  auto const dx2 = get_mi_vector(p2->r.p, p3->r.p);
+  auto const dx3 = get_mi_vector(p4->r.p, p3->r.p);
 
   // Get normals on triangle; pointing outwards by definition of indices
   // sequence
-  double n1[3], n2[3];
-  vector_product(dx1, dx2, n1);
-  vector_product(dx1, dx3, n2);
-
-  // Wolfgang here had a minus. It seems to work, so leave it in
-  n2[0] = -1 * n2[0];
-  n2[1] = -1 * n2[1];
-  n2[2] = -1 * n2[2];
+  auto n1 = vector_product(dx1, dx2);
+  auto n2 = vector_product(dx3, dx1);
 
   // Get 2*area of triangles out of the magnitude of the resulting normals and
   // make the latter unity
-  const double Ai = sqrt(n1[0] * n1[0] + n1[1] * n1[1] + n1[2] * n1[2]);
-  n1[0] = n1[0] / Ai;
-  n1[1] = n1[1] / Ai;
-  n1[2] = n1[2] / Ai;
+  auto const Ai = n1.norm();
+  n1 /= Ai;
 
-  const double Aj = sqrt(n2[0] * n2[0] + n2[1] * n2[1] + n2[2] * n2[2]);
-  n2[0] = n2[0] / Aj;
-  n2[1] = n2[1] / Aj;
-  n2[2] = n2[2] / Aj;
+  auto const Aj = n2.norm();
+  n2 /= Aj;
 
   // Get the prefactor for the force term
-  double sc = scalar(n1, n2);
+  double sc = n1 * n2;
   if (sc > 1.0)
     sc = 1.0;
 
   // Get theta as angle between normals
   double theta = acos(sc);
 
-  double direc[3];
-  vector_product(n1, n2, direc);
-  const double desc = scalar(dx1, direc);
+  auto const direc = vector_product(n1, n2);
+  const double desc = (dx1 * direc);
```

Comment:
Death to the traitor! :point_up_2: :point_up: :point_up_2: :point_up:

Reply:
Didn't we agree not to enforce this because there is no tool to check it? ^^

Reply:
but it makes me happy :smiling_imp: that's why I approved nonetheless